service/app/controllers/skills/UserSkillController.java

```diff
@@ -1,17 +1,35 @@
 package controllers.skills;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import controllers.BaseController;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
 import org.sunbird.common.models.util.ActorOperations;
+import org.sunbird.common.models.util.JsonKey;
 import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
 import play.libs.F.Promise;
 import play.mvc.Result;
 
 public class UserSkillController extends BaseController {
 
   public Promise<Result> updateSkill() {
     try {
-
-      Request reqObj = createAndInitRequest(ActorOperations.UPDATE_SKILL.getValue());
+      JsonNode bodyJson = request().body().asJson();
+      Request reqObj = createAndInitRequest(ActorOperations.UPDATE_SKILL.getValue(), bodyJson);
+      String userId = (String) reqObj.getRequest().get("userID");
+      if (StringUtils.isBlank(userId)) {
+        throw new ProjectCommonException(
+            ResponseCode.mandatoryParamsMissing.getErrorCode(),
+            ResponseCode.mandatoryParamsMissing.getErrorMessage(),
+            ResponseCode.mandatoryParamsMissing.getResponseCode());
+      }
+      if (!userId.equals(ctx().flash().get(JsonKey.USER_ID))) {
+        throw new ProjectCommonException(
+            ResponseCode.invalidParameterValue.getErrorCode(),
+            ResponseCode.invalidParameterValue.getErrorMessage(),
```

Comment:
Placeholders are not being passed in case of invalidParameterValue.

String INVALID_PARAMETER_VALUE = "Invalid value {0} for parameter {1}. Please provide a valid value.";